compiler/test/dotty/tools/utils.scala

```diff
@@ -43,3 +43,24 @@ def assertThrows[T <: Throwable: ClassTag](p: T => Boolean)(body: => Any): Unit
     case failed: T       => throw AssertionError(s"Exception failed check: $failed").tap(_.addSuppressed(failed))
     case NonFatal(other) => throw AssertionError(s"Wrong exception: expected ${implicitly[ClassTag[T]]} but was ${other.getClass.getName}").tap(_.addSuppressed(other))
 end assertThrows
+
+def toolArgsFor(files: List[JPath])(using codec: Codec = Codec.UTF8): List[String] =
+  files.flatMap(path => toolArgsParse(readLines(path.toFile)))
```

Comment:
`readLines` is all the lines? Previously I think a lazy prefix. I've wondered about some extra I/O in partest to check filters and what impact it has in CI, especially in aggregate.

Reply:
I often worry that I'm too paranoid about such performance concerns and should look to achieving more by worrying less.  So I was probably feeling brave back when I wrote this patch, changing it to just eagerly read all the lines.  I'll check aggregate CI time and see if it looks bad.

Reply:
Didn't mean to nitpick, BTW. I have no idea if a) the I/O matters once everything is cached somewhere, and the files are small b) if the extra allocation. I just "tweaked" some code that had been optimized to avoid allocations, and as you know, these optimizations are so "situational".

Reply:
The aggregate looks good to me: picking latest master of running bootstrapped tests took 49m and 42m in this PR, so not double or 1.5x or so.

https://github.com/lampepfl/dotty/runs/4575305597?check_suite_focus=true